modules/casper.js

```diff
@@ -2745,7 +2745,46 @@ function createPage(casper) {
     };
     page.onPageCreated = function onPageCreated(popupPage) {
         casper.emit('popup.created', popupPage);
-        popupPage.onLoadFinished = function onLoadFinished() {
+
+        ['onAlert', 'onCallback', 'onConfirm', 'onConsoleMessage',
+        'onError', 'onFilePicker', 'onInitialized', 'onClosing',
+        'onNavigationRequested', 'onPrompt', 'onResourceReceived',
+        'onResourceError'
+        // 'onPageCreated',
+        // 'onLoadStarted',
+        // 'onLoadFinished',
+        // 'onUrlChanged'
+        // 'onResourceRequested',
+        ].forEach(function (event) {
+             popupPage[event] = casper.page[event];
+        });
+        popupPage.onPageCreated = function (popupBis){
+           page.onPageCreated(popupBis);
+        };
+        popupPage.onResourceRequested = function (requestData, request) {
+            if (casper.page.settings.userAgent !== defaultUserAgent) {
+                if (typeof request.setHeader !== "undefined") {
+                    request.setHeader('User-Agent', casper.page.settings.userAgent);
+                    for (var i = 0, l = requestData.headers.length; i < l; i++) {
+                        if (requestData.headers[i].name === "User-Agent") {
+                            requestData.headers[i].value = casper.page.settings.userAgent;
+                        }
+                    }
+                } else {
+                    casper.warn("The HTTP User-Agent's Header can't be used with popup for older browser(PhantomJS < 2.0.0 and SlimerJS < 0.9.0)");
+                }
+            }
+            page.onResourceRequested(requestData, request);
+        };
+        popupPage.onLoadStarted = function(){
+           casper.emit('load.started');
+        };
+        popupPage.onUrlChanged = function onUrlChanged(url) {
+          casper.log(f('url changed to "%s"', url), "debug");
+        // casper.navigationRequested = false;
```

Comment:
I made a copy paste from page.onUrlChanged : but navigationRequested flag is not adapted to popup.